src/open_inwoner/openzaak/info_objects.py

```diff
@@ -0,0 +1,139 @@
+import logging
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import List, Optional
+
+import requests
+from requests import HTTPError, RequestException, Response
+from zds_client import ClientError
+from zgw_consumers.api_models.base import ZGWModel, factory
+
+from open_inwoner.openzaak.clients import build_client
+from open_inwoner.openzaak.statuses import logger
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class InformatieObject(ZGWModel):
+    url: str
+    identificatie: str
+    bronorganisatie: str
+    creatiedatum: date
+    titel: str
+    vertrouwelijkheidaanduiding: str
+    auteur: str
+    status: str
+    formaat: str
+    taal: str
+    versie: int
+    # beginRegistratie: datetime
+    bestandsnaam: str
+    inhoud: str
+    bestandsomvang: int
+    link: str
+    beschrijving: str
+    ontvangstdatum: str
+    verzenddatum: str
+    # indicatieGebruiksrecht: str
+    ondertekening: dict  # {'soort': '', 'datum': None}
+    integriteit: dict  # {'algoritme': '', 'waarde': '', 'datum': None}
+    informatieobjecttype: str
+    locked: bool
+    # bestandsdelen: List[str]
+
+
+@dataclass
+class ZaakInformatieObject(ZGWModel):
+    url: str
+    informatieobject: str
+    zaak: str
+    # aard_relatie_weergave: str
+    titel: str
+    # beschrijving: str
+    registratiedatum: datetime
+
+
+def fetch_case_information_objects(case_url: str) -> List[ZaakInformatieObject]:
+    client = build_client("zaak")
+
+    if client is None:
+        return []
+
+    try:
+        response = client.list(
+            "zaakinformatieobject",
+            request_kwargs={
+                "params": {"zaak": case_url},
+            },
+        )
+    except RequestException as e:
+        logger.exception("exception while making request", exc_info=e)
+        return []
+    except ClientError as e:
+        logger.exception("exception while making request", exc_info=e)
+        return []
+
+    case_info_objects = factory(ZaakInformatieObject, response)
+
+    return case_info_objects
```

Comment:
This was in another file that got crowded.

Reply:
I agree with option 1. We'll need to coordinate how/when to refactor that.

Reply:
Let's do it in the PR and we will merge it and rebase our branches and be done with this :)

Reply:
I've added the refactoring to the PR. The commit-diff is interesting as it is trying to make sense of code movements.